lib/settings.js

```diff
@@ -118,17 +123,56 @@ async function updateLocationSettings (sim, bundleId, authorized) {
     }
 
     // create the update, and make sure it has sensible values
-    let baseSetting = _.has(plist[0], bundleId) ? plist[0][bundleId] : {};
+    let baseSetting = _.has(plist, bundleId) ? plist[bundleId] : {};
     updates[bundleId] = _.defaults(newClientPrefs, baseSetting);
     updates[bundleId].Executable = updates[bundleId].Executable || '';
     updates[bundleId].Registered = updates[bundleId].Registered || '';
 
-    await update(file, [updates]);
+    await update(file, updates);
   }
 }
 
+async function updateSafariUserSettings (sim, settingSet) {
+  // add extra stuff to UserSettings.plist and EffectiveUserSettings.plist
+  let newUserSettings = {};
+  if (_.has(settingSet, 'WebKitJavaScriptEnabled')) {
+    newUserSettings.safariAllowJavaScript = settingSet.WebKitJavaScriptEnabled;
+  }
+  if (_.has(settingSet, 'WebKitJavaScriptCanOpenWindowsAutomatically')) {
+    newUserSettings.safariAllowPopups = settingSet.WebKitJavaScriptCanOpenWindowsAutomatically;
+  }
+  if (_.has(settingSet, 'WarnAboutFraudulentWebsites')) {
+    newUserSettings.safariForceFraudWarning = !settingSet.WarnAboutFraudulentWebsites;
+  }
+  if (_.size(newUserSettings) > 0) {
+    log.debug('Updating Safari user settings');
+    let userSettingsPerFile = {};
+    let curUserSettings = await readSettings(sim, 'userSettings');
+    // _.each(curUserSettings, function (userSettingSet, file) {
+    for (let [file, userSettingSet] of _.pairs(curUserSettings)) {
+      if (!_.has(userSettingSet, 'restrictedBool')) {
+        userSettingSet.restrictedBool = {};
```

Comment:
might want a comment above here explaining that theres this weird restrictedBool deal that is a magic thing for some reason